### PR TITLE
Add protocol for calibration of pulse to generate state 2

### PR DESCRIPTION
**Pull request by andrea-pasquale**
This PR adds two protocols:

- Qubit spectroscopy EF to estimate the frequency of the 1->2 transition (or the anharmonicity)
- Rabi amplitude EF to calibrate the amplitude necessary to produce state 2 starting from 1

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Comment by andrea-pasquale** [comment]
Tested on qw5q_gold.

image

image


Seems to be working as expected! We should be able to produce state 2 now.

**Review by GabrielePalazzo — COMMENTED** [review]
Code-wise it all seems correct to me. I'll do some tests tomorrow.

**Review by Edoardo-Pedicillo — APPROVED** [review]

File: src/qibocal/protocols/characterization/rabi/ef.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+@dataclass
+class RabiAmplitudeEFResults(Results):
+    """RabiAmplitudeEF outputs."""
+
+    amplitude: dict[QubitId, float]
+    """Drive amplitude for each qubit."""
+    length: dict[QubitId, float]
+    """Drive pulse duration. Same for all qubits."""
+    fitted_parameters: dict[QubitId, dict[str, float]]
+    """Raw fitted parameters."""
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/rabi/ef.py:30]: Could you make `RabiAmplitudeEFResults` inherit from `amplitute.RabiAmplitudeResults`?

Comment (andrea-pasquale) [reply src/qibocal/protocols/characterization/rabi/ef.py]: Done :)

**Review by GabrielePalazzo — APPROVED** [review]

**Pull request merged**